op-challenger/fault/monitor.go

```diff
@@ -143,25 +150,20 @@ func (m *gameMonitor) fetchOrCreateGamePlayer(gameData FaultDisputeGame) (gamePl
 func (m *gameMonitor) MonitorGames(ctx context.Context) error {
 	m.logger.Info("Monitoring fault dispute games")
 
-	blockNum := uint64(0)
+	headers, sub, err := m.headSubscriber()
+	if err != nil {
+		return fmt.Errorf("failed to subscribe to new headers: %w", err)
+	}
+
 	for {
 		select {
+		case err := <-sub.Err():
+			return fmt.Errorf("subscription error: %w", err)
```

Comment:
I think we need to log this and reconnect the subscription. Its pretty common for a subscription to get disconnected and we don't want to crash in that situation.

Reply:
Agreed, just wanted to elicit your review for the overall setup first before going too deep in the implementation.